lnrpc/routerrpc/router.proto

```diff
@@ -335,6 +335,136 @@ message BuildRouteResponse {
     lnrpc.Route route = 1;
 }
 
+
+message SubscribeHtlcEventsRequest{}
+
+message HtlcEvent{
+    /**
+    The short channel id that the incoming htlc arrived at our node on. This
+    value is zero for sends.
+    */
+    uint64 incoming_channel_id = 1;
+
+    /**
+    The short channel id that the outgoing htlc left our node on. This value
+    is zero for receives.
+    */
+    uint64 outgoing_channel_id = 2;
+
+    /**
+    Incoming id is the index of the incoming htlc in the incoming channel.
+    This value is zero for sends.
+    */
+    uint64 incoming_htlc_id = 3;
+
+    /**
+    Outgoing id is the index of the outgoing htlc in the outgoing channel.
+    This value is zero for receives.
+    */
+    uint64 outgoing_htlc_id = 4;
+
+    /**
+    Timestamp is the time (unix epoch offset in seconds) that the event
+    occurred.
+    */
+    uint64 timestamp = 5;
+
+    enum EventType{
+        UNKNOWN = 0;
+        SEND = 1;
+        RECEIVE = 2;
+        FORWARD = 3;
+    }
+
+    /**
+    The event type indicates whether the htlc was part of a send, receive or
+    forward.
+    */
+    EventType event_type = 6;
+
+    /*
+    Event contains the htlc event that was processed. These are served on a
+    best-effort basis; events are not persisted, delivery is not guaranteed
+    (in the event of a crash in the switch, forward events may be lost) and
+    some events may be replayed upon restart. Events consumed from this package
+    should be de-duplicated by the htlc's unique combination of incoming and
+    outgoing channel id and htlc id. [EXPERIMENTAL]
```

Comment:
Maybe this could be moved one level up. It applies to all of the `HtlcEvent` proto message?